src/lib/mongoBackend/mongoRegisterContext.cpp

```diff
@@ -109,9 +109,10 @@ HttpStatusCode mongoRegisterContext
     return SccOk;
   }
 
-  const char* colName = getRegistrationsCollectionName(tenant).c_str();
+  const std::string    colName  = getRegistrationsCollectionName(tenant);
+  const mongo::BSONObj bson     = BSON("_id" << id << REG_SERVICE_PATH << sPath);
 
-  if (!collectionFindOne(colName, BSON("_id" << id << REG_SERVICE_PATH << sPath), &reg, &err))
+  if (!collectionFindOne(colName, bson, &reg, &err))
```

Comment:
Here is the change that fixes the local problem I had with unittests.
The problem I saw was that `colName` was overwritten with "servicePath", a text string that belongs to the BSON() macro call.
I created a help variable for the BSON stuff and the problem went away.
Not 100% sure this is the fix but ... that BSON call as a parameter (accepted as a const `mongo::BSONObj&` by `collectionFindOne`) is not very nice ...